applications/physics/cosmology/cosmoflow/cosmoflow_dataset.py

```diff
@@ -0,0 +1,30 @@
+import numpy as np
+from glob import glob
+from lbann.util.data import Sample, SampleDims, Dataset, DistConvDataset
+import h5py as h5
+import os
+
+    
+class CosmoFlowDataset(DistConvDataset):
+    def __init__(self, data_dir, input_width, num_secrets):
+        self.data_dir = data_dir
+        self.input_width = input_width
+        self.num_secrets = num_secrets
+        self.samples = glob(os.path.join(data_dir, '*.hdf5'))
+    
+    def __len__(self):
+        return len(self.samples)
+    
+    def __getitem__(self, index) -> Sample:
+        data = h5.File(self.samples[index], 'r')
+        slice_width = self.input_width // self.num_io_partitions
+        slice_ind = self.rank % self.num_io_partitions
+        full = data['full'][:,
```

Comment:
@fiedorowicz1 Is this actually selecting the data set / hyperslab that you are expecting?  Looking at https://docs.h5py.org/en/stable/high/dataset.html, it seems that there is a dataset object that I don't see created here.